hail/src/main/scala/is/hail/expr/types/encoded/EType.scala

```diff
@@ -245,9 +246,9 @@ object EType {
     }
   }
 
-  def defaultFromPType(pt: PType): EType = defaultFromPType(pt, pt.required)
-
-  def defaultFromPType(pt: PType, required: Boolean): EType = {
+  def defaultFromPType(pt: PType,
+    useTransposedArrayOfStructs: Boolean = true
```

Comment:
we already check the flags in the match, why do we need this flag? I don't really like this model because it won't scale to many interesting ETypes.

Reply:
We will probably need hinting at some future point, but we don't need it yet honestly.